embedded/sql/catalog_test.go

```diff
@@ -130,3 +132,138 @@ func TestEncodeRawValueAsKey(t *testing.T) {
 		}
 	})
 }
+
+func TestCatalogTableLength(t *testing.T) {
+	// db := newCatalog(nil)
+
+	st, err := store.Open(t.TempDir(), store.DefaultOptions())
+	require.NoError(t, err)
+	defer closeStore(t, st)
+
+	engine, err := NewEngine(st, DefaultOptions().WithPrefix(sqlPrefix))
+	require.NoError(t, err)
+
+	totalTablesCount := uint32(0)
+
+	for _, v := range []string{"table1", "table2", "table3"} {
+		_, _, err = engine.Exec(
+			context.Background(), nil,
+			`
+			CREATE TABLE `+v+` (
+				id INTEGER AUTO_INCREMENT,
+				PRIMARY KEY(id)
+			)`, nil)
+		require.NoError(t, err)
+		totalTablesCount++
+	}
+
+	t.Run("table count should be 3 on catalog reload", func(t *testing.T) {
+		for i := 0; i < 3; i++ {
+			tx, err := engine.NewTx(context.Background(), DefaultTxOptions())
+			require.NoError(t, err)
+			defer tx.Cancel()
+
+			require.Equal(t, totalTablesCount, tx.catalog.tableCount)
+		}
+	})
+
+	t.Run("table count should not increase on adding existing table", func(t *testing.T) {
+		tx, err := engine.NewTx(context.Background(), DefaultTxOptions())
+		require.NoError(t, err)
+		defer tx.Cancel()
+		catlog := tx.catalog
+
+		for _, v := range []string{"table1", "table2", "table3"} {
+			_, err := catlog.newTable(v, []*ColSpec{{colName: "id", colType: IntegerType}})
```

Comment:
it may be better to create from a sql statement as the actual creation is triggered from there.

Also it would be useful to have a transaction creating a table but being cancelled instead of committed, which may not increase the counter.

Additionally, check the case of two concurrent transactions creating tables

Reply:
Mentioned the issue here

https://github.com/orgs/codenotary/projects/10?pane=issue&itemId=25297372